FC Updates

dwomble opened the pull request:
Added Cargo and Locker for fleet carrier.

Closes #297.

aussig commented on bgstally/fleetcarrier.py:
```diff
@@ -331,15 +376,20 @@ def _get_items(self, category: FleetCarrierItemType = None) -> tuple[list|None,
                 # Lookup name and display name are the same for commodities as we are not passed localised name from CAPI. We
                 # convert the display name later
-                return self.commodities_selling, 'name', 'name', 'stock'
+                return self.commodities_selling, 'name', 'locName', 'stock'
```
Did you find that this has changed in the CAPI response? I had this fetching `name` previously as `locName` was not included for Commodities being bought or sold.

dwomble replied:
No, but I generalized the solution. Rather than specifically coding for types that have or do not have a localised name I check if a given record has a localised name and if so use it. Otherwise I look it up. Looks like I messed up and didn't put that code in. I'll do so now.

aussig commented on bgstally/fleetcarrier.py:
```diff
@@ -331,15 +376,20 @@ def _get_items(self, category: FleetCarrierItemType = None) -> tuple[list|None,
+                return self.commodities_buying, 'name', 'locName', 'outstanding'
+            case FleetCarrierItemType.CARGO:
+                # Return cargo items
+                return get_by_path(self.data, ['cargo'], []), 'commodity', 'locName', 'qty'
```
Should this not simply be:

```python
return get_by_path(self.cargo, 'commodity', 'locName', 'qty'
```

To follow the same pattern?  `self.cargo` should be initialised by this point.

aussig commented on bgstally/fleetcarrier.py:
```diff
@@ -331,15 +376,20 @@ def _get_items(self, category: FleetCarrierItemType = None) -> tuple[list|None,
+                return get_by_path(self.data, ['cargo'], []), 'commodity', 'locName', 'qty'
+            case FleetCarrierItemType.LOCKER:
+                # Return locker items
+                return get_by_path(self.data, ['carrierLocker'], []), 'name', 'locName', 'quantity'
```
Should this not simply be:

```python
return get_by_path(self.mats, 'commodity', 'locName', 'qty'
```

To follow the same pattern?  `self.mats` should be initialised by this point.

dwomble replied:
Yes, I put this in temporarily until I added all the other code to create the list etc. etc. and forgot to replace it. TBH it's more memory efficient and simpler to do it this way.

aussig commented on bgstally/fleetcarrier.py:
```diff
@@ -331,15 +376,20 @@ def _get_items(self, category: FleetCarrierItemType = None) -> tuple[list|None,
                 # Lookup name and display name are the same for commodities as we are not passed localised name from CAPI. We
                 # convert the display name later
-                return self.commodities_buying, 'name', 'name', 'outstanding'
+                return self.commodities_buying, 'name', 'locName', 'outstanding'
```
Same comment as above.

aussig commented on bgstally/fleetcarrier.py:
```diff
@@ -196,25 +211,54 @@ def get_items_plaintext(self, category: FleetCarrierItemType|None = None) -> str
-
-        return result
+        try:
+            #Debug.logger.error(f"{category}")
```
Lose the commented debug line.

aussig commented on bgstally/fleetcarrier.py:
```diff
@@ -196,25 +211,54 @@ def get_items_plaintext(self, category: FleetCarrierItemType|None = None) -> str
+                    result += f"{key} x {value}\n"
+                return result
+
+            if category == FleetCarrierItemType.LOCKER:
```
Add a comment as to why LOCKER items are also a special case (you've noted why CARGO is a special case above).

aussig commented on bgstally/fleetcarrier.py:
```diff
@@ -196,25 +211,54 @@ def get_items_plaintext(self, category: FleetCarrierItemType|None = None) -> str
+            if category == FleetCarrierItemType.LOCKER:
+                for type in items:
+                    result += f"{type.title()}:\n"
+                    #Debug.logger.error(items[type])
```
Lose the commented debug line.

aussig commented on bgstally/fleetcarrier.py:
```diff
@@ -196,25 +211,54 @@ def get_items_plaintext(self, category: FleetCarrierItemType|None = None) -> str
+                    #Debug.logger.error(items[type])
+                    items[type] = sorted(items[type], key=lambda x: x[display_name_key])
+                    for item in items[type]:
+                        #if int(item[quantity_key]) > 0: # This one includes zero quantities for some reason
```
Should either be removed or reinstated.

dwomble replied:
Yes, commented out because I wasn't sure to include or exclude. I think include.

aussig commented on bgstally/state.py:
```diff
@@ -31,13 +31,17 @@ def load(self):
         self.EnableOverlaySystem:tk.StringVar = tk.StringVar(value=config.get_str('BGST_EnableOverlaySystem', default=CheckStates.STATE_ON))
         self.EnableOverlayWarning:tk.StringVar = tk.StringVar(value=config.get_str('BGST_EnableOverlayWarning', default=CheckStates.STATE_ON))
         self.EnableOverlayCMDR:tk.StringVar = tk.StringVar(value=config.get_str('BGST_EnableOverlayCMDR', default=CheckStates.STATE_ON))
-        self.EnableOverlayObjectives:tk.StringVar = tk.StringVar(value=config.get_str('BGST_EnableOverlayObjectives', default=CheckStates.STATE_ON))
```
It looks like you didn't pull the latest code from `develop` into your fork.  This line needs to be reinstated, as well as a number of others below, which will come in from `develop`.

dwomble replied:
I specifically did a pull before creating the PR. :(

aussig commented on bgstally/state.py:
```diff
@@ -31,13 +31,17 @@ def load(self):
         self.DiscordActivity:tk.StringVar = tk.StringVar(value=config.get_str('BGST_DiscordActivity', default=DiscordActivity.BOTH))
-        self.DiscordFleetCarrier: tk.StringVar = tk.StringVar(value=config.get_str('BGST_DiscordFleetCarrier', default=DiscordFleetCarrier.BOTH))
-        self.DiscordAvatarURL:tk.StringVar = tk.StringVar(value=config.get_str('BGST_DiscordAvatarURL', default=""))
+        #self.DiscordFleetCarrier: tk.StringVar = tk.StringVar(value=config.get_str('BGST_DiscordFleetCarrier', default=DiscordFleetCarrier.BOTH))
```
Lose this old state setting.

aussig commented on bgstally/state.py:
```diff
@@ -31,13 +31,17 @@ def load(self):
-        self.DiscordFleetCarrier: tk.StringVar = tk.StringVar(value=config.get_str('BGST_DiscordFleetCarrier', default=DiscordFleetCarrier.BOTH))
-        self.DiscordAvatarURL:tk.StringVar = tk.StringVar(value=config.get_str('BGST_DiscordAvatarURL', default=""))
+        #self.DiscordFleetCarrier: tk.StringVar = tk.StringVar(value=config.get_str('BGST_DiscordFleetCarrier', default=DiscordFleetCarrier.BOTH))
+        self.FcSellingCommodities:tk.StringVar = tk.StringVar(value=config.get_str('FcSellingCommodities', default=CheckStates.STATE_ON))
```
All BGS-Tally state should be prefixed with `BGST_`. This is a recommendation by the EDMC team to minimise conflicts between state from different plugins.

aussig commented on bgstally/state.py:
```diff
@@ -100,13 +110,17 @@ def save(self):
         config.set('BGST_EnableOverlaySystem', self.EnableOverlaySystem.get())
         config.set('BGST_EnableOverlayWarning', self.EnableOverlayWarning.get())
         config.set('BGST_EnableOverlayCMDR', self.EnableOverlayCMDR.get())
-        config.set('BGST_EnableOverlayObjectives', self.EnableOverlayObjectives.get())
```
Reinstate.

aussig commented on bgstally/state.py:
```diff
@@ -100,13 +110,17 @@ def save(self):
         config.set('BGST_DetailedTrade', self.DetailedTrade.get())
         config.set('BGST_DiscordActivity', self.DiscordActivity.get())
-        config.set('BGST_DiscordFleetCarrier', self.DiscordFleetCarrier.get())
-        config.set('BGST_DiscordAvatarURL', self.DiscordAvatarURL.get())
```
Reinstate.

aussig commented on bgstally/state.py:
```diff
@@ -100,13 +110,17 @@ def save(self):
         config.set('BGST_DiscordActivity', self.DiscordActivity.get())
-        config.set('BGST_DiscordFleetCarrier', self.DiscordFleetCarrier.get())
-        config.set('BGST_DiscordAvatarURL', self.DiscordAvatarURL.get())
+        #config.set('BGST_DiscordFleetCarrier', self.DiscordFleetCarrier.get())
```
Lose this old state setting.

aussig commented on bgstally/windows/fleetcarrier.py:
```diff
@@ -54,8 +54,18 @@ def show(self):
 
-        ttk.Label(frm_items, text=_("Selling Materials"), font=FONT_HEADING_2).grid(row=current_row, column=0, sticky=tk.W) # LANG: Label on carrier window
-        ttk.Label(frm_items, text=_("Buying Materials"), font=FONT_HEADING_2).grid(row=current_row, column=1, sticky=tk.W) # LANG: Label on carrier window
+        #ttk.Checkbutton(frm_discordoptions, text=_("Abbreviate Faction Names"), variable=self.bgstally.state.AbbreviateFactionNames, onvalue=CheckStates.STATE_ON, offvalue=CheckStates.STATE_OFF, command=partial(self._option_change, activity)).grid(row=current_row, column=0, padx=10, sticky=tk.W); current_row += 1 # LANG: Checkbox label
+        dummy = ""
+        a = ""
+        b = ""
```
I think these 4 lines can go?

aussig commented on bgstally/windows/fleetcarrier.py:
```diff
@@ -54,8 +54,18 @@ def show(self):
+        dummy = ""
+        a = ""
+        b = ""
+        s = ttk.Style() # LANG: Checkbox label
```
1. Add type hint
2. Remove `# LANG:` comment from this line - that is additional info for translators when there is a translatable string on the same line, but there is no translatable string on this line.

aussig commented on bgstally/windows/fleetcarrier.py:
```diff
@@ -83,8 +93,11 @@ def show(self):
 
-        ttk.Label(frm_items, text=_("Selling Commodities"), font=FONT_HEADING_2).grid(row=current_row, column=0, sticky=tk.W) # LANG: Label on carrier window
-        ttk.Label(frm_items, text=_("Buying Commodities"), font=FONT_HEADING_2).grid(row=current_row, column=1, sticky=tk.W) # LANG: Label on carrier window
+        ttk.Checkbutton(frm_items, text=_("Selling Commodities"), style='This.TCheckbutton', variable=self.bgstally.state.FcSellingCommodities, onvalue=CheckStates.STATE_ON, \
+                        offvalue=CheckStates.STATE_OFF, command=partial(self.bgstally.state.FcSellingCommodities.set)).grid(row=current_row, column=0, padx=10, sticky=tk.W)
+
+        ttk.Checkbutton(frm_items, text=_("Buying Commodities"), style='This.TCheckbutton', variable=self.bgstally.state.FcBuyingCommodities, onvalue=CheckStates.STATE_ON, \
+                        offvalue=CheckStates.STATE_OFF, command=partial(self.bgstally.state.FcBuyingCommodities.set)).grid(row=current_row, column=1, padx=10, sticky=tk.W)
```
_This makes me think I should be moving to ttk styles throughout :)  My problem, not yours!_

_We also need translation hints for these lines, like this: ` # LANG: Label on carrier window` but my hacked-together hit extractor can't cope with continuation lines using `\` where the translatable string is on a different line to the hint.  Also my problem, not yours._

aussig reviewed (changes requested):
Thank you very much for this PR. I've reviewed and commented.

dwomble commented:
Thanks,

I'm also investigating an issue with it not updating. I suspect it's
because we aren't calling the CAPI often enough. Cargo can change without
any buy, sell, or marketplace events.This can lead to the data being very
out of date.

Derek

On Wed, Mar 19, 2025 at 5:39 AM Aussi ***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
>
> Looking good, just one final debug line to remove
> ------------------------------
>
> In bgstally/fleetcarrier.py
> <https://github.com/aussig/BGS-Tally/pull/301#discussion_r2003220534>:
>
> >              case FleetCarrierItemType.COMMODITIES_BUYING:
>                  # Lookup name and display name are the same for commodities as we are not passed localised name from CAPI. We
>                  # convert the display name later
> -                return self.commodities_buying, 'name', 'name', 'outstanding'
> +                return self.commodities_buying, 'name', 'locName', 'outstanding'
> +            case FleetCarrierItemType.CARGO:
> +                # Return cargo items
> +                return self.cargo, 'commodity', 'locName', 'qty'
> +            case FleetCarrierItemType.LOCKER:
> +                # Return locker items
> +                Debug.logger.error(f"Locker: {get_by_path(self.data, ['carrierLocker'], [])} ")
>
> Remove error log
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/aussig/BGS-Tally/pull/301#pullrequestreview-2698245589>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAN5K2TTOGV5FL42WDEOSOL2VFQPNAVCNFSM6AAAAABZHDPJ4CVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDMOJYGI2DKNJYHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

aussig commented:
> Thanks, I'm also investigating an issue with it not updating. I suspect it's because we aren't calling the CAPI often enough. Cargo can change without any buy, sell, or marketplace events.This can lead to the data being very out of date. Derek

The CAPI calls are limited (in EDMC) to one every 15 minutes maximum, and only when opening the carrier screen in-game. If you suspect a bug then perhaps raise a separate issue with your findings and we can tackle outside this PR.

dwomble commented:
"and only when opening the carrier screen in-game" – there's the issue. If
you use the inventory or if someone else buys from or sells to your carrier
the cargo & materials will change without you visiting the screen.

I'll see if I can get them to do it more frequently. With groups working on
loading and unloading each others' carriers for colonization, seeing
progress without visiting your carrier is much more valuable.

On Wed, Mar 19, 2025 at 8:46 AM Aussi ***@***.***> wrote:

> Thanks, I'm also investigating an issue with it not updating. I suspect
> it's because we aren't calling the CAPI often enough. Cargo can change
> without any buy, sell, or marketplace events.This can lead to the data
> being very out of date. Derek
>
> The CAPI calls are limited (in EDMC) to one every 15 minutes maximum, and
> only when opening the carrier screen in-game. If you suspect a bug then
> perhaps raise a separate issue with your findings and we can tackle outside
> this PR.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/aussig/BGS-Tally/pull/301#issuecomment-2737138933>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAN5K2RMFYXSQLYNBZOIE732VGGNJAVCNFSM6AAAAABZHDPJ4CVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMZXGEZTQOJTGM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: aussig]*aussig* left a comment (aussig/BGS-Tally#301)
> <https://github.com/aussig/BGS-Tally/pull/301#issuecomment-2737138933>
>
> Thanks, I'm also investigating an issue with it not updating. I suspect
> it's because we aren't calling the CAPI often enough. Cargo can change
> without any buy, sell, or marketplace events.This can lead to the data
> being very out of date. Derek
>
> The CAPI calls are limited (in EDMC) to one every 15 minutes maximum, and
> only when opening the carrier screen in-game. If you suspect a bug then
> perhaps raise a separate issue with your findings and we can tackle outside
> this PR.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/aussig/BGS-Tally/pull/301#issuecomment-2737138933>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAN5K2RMFYXSQLYNBZOIE732VGGNJAVCNFSM6AAAAABZHDPJ4CVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMZXGEZTQOJTGM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

dwomble commented:
Could BGS-Tally call the trigger code when the fleetcarrier window is
opened? The cooldown could still be applied to prevent excessive calls.

On Wed, Mar 19, 2025 at 9:30 AM Aussi ***@***.***> wrote:

> "and only when opening the carrier screen in-game" – there's the issue. If
> you use the inventory or if someone else buys from or sells to your carrier
> the cargo & materials will change without you visiting the screen. I'll see
> if I can get them to do it more frequently. With groups working on loading
> and unloading each others' carriers for colonization, seeing progress
> without visiting your carrier is much more valuable.
>
> That's actually my EDMC code :) - I added it specifically to allow this
> feature in BGS-Tally, as I wanted to avoid creating a whole CAPI process
> myself.
>
> The call is triggered here
> <https://github.com/EDCD/EDMarketConnector/blob/2f3f7262da81f4892972560822662e9205c7997c/EDMarketConnector.py#L1564>,
> it's handled here
> <https://github.com/EDCD/EDMarketConnector/blob/2f3f7262da81f4892972560822662e9205c7997c/EDMarketConnector.py#L1066>
> and the cooldown is applied here
> <https://github.com/EDCD/EDMarketConnector/blob/2f3f7262da81f4892972560822662e9205c7997c/EDMarketConnector.py#L1099>
>
> I was told at the time that it needed a long cooldown to ensure it didn't
> cause too many CAPI calls.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/aussig/BGS-Tally/pull/301#issuecomment-2737305292>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAN5K2QU5ELBCSHT2QEQPVT2VGLRHAVCNFSM6AAAAABZHDPJ4CVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMZXGMYDKMRZGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: aussig]*aussig* left a comment (aussig/BGS-Tally#301)
> <https://github.com/aussig/BGS-Tally/pull/301#issuecomment-2737305292>
>
> "and only when opening the carrier screen in-game" – there's the issue. If
> you use the inventory or if someone else buys from or sells to your carrier
> the cargo & materials will change without you visiting the screen. I'll see
> if I can get them to do it more frequently. With groups working on loading
> and unloading each others' carriers for colonization, seeing progress
> without visiting your carrier is much more valuable.
>
> That's actually my EDMC code :) - I added it specifically to allow this
> feature in BGS-Tally, as I wanted to avoid creating a whole CAPI process
> myself.
>
> The call is triggered here
> <https://github.com/EDCD/EDMarketConnector/blob/2f3f7262da81f4892972560822662e9205c7997c/EDMarketConnector.py#L1564>,
> it's handled here
> <https://github.com/EDCD/EDMarketConnector/blob/2f3f7262da81f4892972560822662e9205c7997c/EDMarketConnector.py#L1066>
> and the cooldown is applied here
> <https://github.com/EDCD/EDMarketConnector/blob/2f3f7262da81f4892972560822662e9205c7997c/EDMarketConnector.py#L1099>
>
> I was told at the time that it needed a long cooldown to ensure it didn't
> cause too many CAPI calls.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/aussig/BGS-Tally/pull/301#issuecomment-2737305292>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAN5K2QU5ELBCSHT2QEQPVT2VGLRHAVCNFSM6AAAAABZHDPJ4CVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMZXGMYDKMRZGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

aussig commented:
> Could BGS-Tally call the trigger code when the fleetcarrier window is opened? The cooldown could still be applied to prevent excessive calls.

That's how it's currently working - the CAPI is called when the FC window is opened, unless it's on cooldown.

If you want to increase the frequency, then it would be a request to the EDMC team to change [this value](https://github.com/EDCD/EDMarketConnector/blob/2f3f7262da81f4892972560822662e9205c7997c/companion.py#L52). I wouldn't want to call internal "private" EDMC functions to bypass the cooldown as that would be highly sensitive to change.

dwomble commented:
For sure, Im not suggesting bypassing the c/d. When you say “fleet carrier
window” do you mean the BGS-tally one or the in game one?

I’m fairly sure it didn’t happen for me until I opened the in-game one.


On Wed, Mar 19, 2025 at 12:56 PM Aussi ***@***.***> wrote:

> Could BGS-Tally call the trigger code when the fleetcarrier window is
> opened? The cooldown could still be applied to prevent excessive calls.
>
> That's how it's currently working - the CAPI is called when the FC window
> is opened, unless it's on cooldown.
>
> If you want to increase the frequency, then it would be a request to the
> EDMC team to change this value
> <https://github.com/EDCD/EDMarketConnector/blob/2f3f7262da81f4892972560822662e9205c7997c/companion.py#L52>.
> I wouldn't want to call internal "private" EDMC functions to bypass the
> cooldown as that would be highly sensitive to change.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/aussig/BGS-Tally/pull/301#issuecomment-2737945565>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAN5K2T6UY3IQXSG46QATDT2VHDWBAVCNFSM6AAAAABZHDPJ4CVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMZXHE2DKNJWGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: aussig]*aussig* left a comment (aussig/BGS-Tally#301)
> <https://github.com/aussig/BGS-Tally/pull/301#issuecomment-2737945565>
>
> Could BGS-Tally call the trigger code when the fleetcarrier window is
> opened? The cooldown could still be applied to prevent excessive calls.
>
> That's how it's currently working - the CAPI is called when the FC window
> is opened, unless it's on cooldown.
>
> If you want to increase the frequency, then it would be a request to the
> EDMC team to change this value
> <https://github.com/EDCD/EDMarketConnector/blob/2f3f7262da81f4892972560822662e9205c7997c/companion.py#L52>.
> I wouldn't want to call internal "private" EDMC functions to bypass the
> cooldown as that would be highly sensitive to change.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/aussig/BGS-Tally/pull/301#issuecomment-2737945565>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAN5K2T6UY3IQXSG46QATDT2VHDWBAVCNFSM6AAAAABZHDPJ4CVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMZXHE2DKNJWGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

aussig commented:
> For sure, Im not suggesting bypassing the c/d. When you say “fleet carrier window” do you mean the BGS-tally one or the in game one? I’m fairly sure it didn’t happen for me until I opened the in-game one.

It's the in-game one. EDMC spots the journal entry and kicks off the CAPI request.

dwomble commented:
Right. What I was consider was triggering based on the BGS-Tally window as
well, but including the cooldown, we certainly don't want to bypass that
part of it.

The other option is we just treat it as a limitation... If you want your
cargo updated you have to visit your ingame carrier management screen.

Derek

On Wed, Mar 19, 2025 at 1:10 PM Aussi ***@***.***> wrote:

> For sure, Im not suggesting bypassing the c/d. When you say “fleet carrier
> window” do you mean the BGS-tally one or the in game one? I’m fairly sure
> it didn’t happen for me until I opened the in-game one.
>
> It's the in-game one. EDMC spots the journal entry and kicks off the CAPI
> request.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/aussig/BGS-Tally/pull/301#issuecomment-2737983605>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAN5K2UMNOM742UGHHAEVRT2VHFKRAVCNFSM6AAAAABZHDPJ4CVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMZXHE4DGNRQGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: aussig]*aussig* left a comment (aussig/BGS-Tally#301)
> <https://github.com/aussig/BGS-Tally/pull/301#issuecomment-2737983605>
>
> For sure, Im not suggesting bypassing the c/d. When you say “fleet carrier
> window” do you mean the BGS-tally one or the in game one? I’m fairly sure
> it didn’t happen for me until I opened the in-game one.
>
> It's the in-game one. EDMC spots the journal entry and kicks off the CAPI
> request.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/aussig/BGS-Tally/pull/301#issuecomment-2737983605>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAN5K2UMNOM742UGHHAEVRT2VHFKRAVCNFSM6AAAAABZHDPJ4CVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMZXHE4DGNRQGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

aussig commented:
> Right. What I was consider was triggering based on the BGS-Tally window as well, but including the cooldown, we certainly don't want to bypass that part of it. The other option is we just treat it as a limitation... If you want your cargo updated you have to visit your ingame carrier management screen. 

Remember that the CAPI call can take some time to come back, so even if we trigger on window open, the window wouldn't have up to date info until a little later anyway, so would need to either be opened twice, or live refreshed.  I suspect accepting the limitation will be the approach for now, unless we can persuade Rixxan to lower that timeout.

dwomble commented:
Fair. I don’t think the cooldown is the problem per se, it needs to check
on more events and/or on a schedule, though I’m not sure which events.

On Wed, Mar 19, 2025 at 1:34 PM Aussi ***@***.***> wrote:

> Right. What I was consider was triggering based on the BGS-Tally window as
> well, but including the cooldown, we certainly don't want to bypass that
> part of it. The other option is we just treat it as a limitation... If you
> want your cargo updated you have to visit your ingame carrier management
> screen.
>
> Remember that the CAPI call can take some time to come back, so even if we
> trigger on window open, the window wouldn't have up to date info until a
> little later anyway, so would need to either be opened twice, or live
> refreshed. I suspect accepting the limitation will be the approach for now,
> unless we can persuade Rixxan to lower that timeout.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/aussig/BGS-Tally/pull/301#issuecomment-2738040430>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAN5K2XXJMPOKOA4A77HD6L2VHIFLAVCNFSM6AAAAABZHDPJ4CVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMZYGA2DANBTGA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: aussig]*aussig* left a comment (aussig/BGS-Tally#301)
> <https://github.com/aussig/BGS-Tally/pull/301#issuecomment-2738040430>
>
> Right. What I was consider was triggering based on the BGS-Tally window as
> well, but including the cooldown, we certainly don't want to bypass that
> part of it. The other option is we just treat it as a limitation... If you
> want your cargo updated you have to visit your ingame carrier management
> screen.
>
> Remember that the CAPI call can take some time to come back, so even if we
> trigger on window open, the window wouldn't have up to date info until a
> little later anyway, so would need to either be opened twice, or live
> refreshed. I suspect accepting the limitation will be the approach for now,
> unless we can persuade Rixxan to lower that timeout.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/aussig/BGS-Tally/pull/301#issuecomment-2738040430>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAN5K2XXJMPOKOA4A77HD6L2VHIFLAVCNFSM6AAAAABZHDPJ4CVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMZYGA2DANBTGA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

aussig commented:
Ok cool, I think you just have one tweak to make and I'll accept the PR.

dwomble commented:
Done. Also fixed a bug.


On Thu, Mar 20, 2025 at 1:18 AM Aussi ***@***.***> wrote:

> Ok cool, I think you just have one tweak to make and I'll accept the PR.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/aussig/BGS-Tally/pull/301#issuecomment-2739538261>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAN5K2XRUNFDJFNWQZUJ64L2VJ2WLAVCNFSM6AAAAABZHDPJ4CVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMZZGUZTQMRWGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: aussig]*aussig* left a comment (aussig/BGS-Tally#301)
> <https://github.com/aussig/BGS-Tally/pull/301#issuecomment-2739538261>
>
> Ok cool, I think you just have one tweak to make and I'll accept the PR.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/aussig/BGS-Tally/pull/301#issuecomment-2739538261>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAN5K2XRUNFDJFNWQZUJ64L2VJ2WLAVCNFSM6AAAAABZHDPJ4CVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMZZGUZTQMRWGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
